src/app/OperationalSessionSetup.cpp

```diff
@@ -553,27 +576,38 @@ CHIP_ERROR OperationalSessionSetup::ScheduleSessionSetupReattempt(System::Clock:
     return err;
 }
 
-void OperationalSessionSetup::TrySetupAgain(System::Layer * systemLayer, void * state)
+void OperationalSessionSetup::CancelSessionSetupReattempt()
 {
-    auto * self = static_cast<OperationalSessionSetup *>(state);
-
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    // If we can't get a system layer, there is no way for us to cancel things
+    // at this point, but hopefully that's because everything is torn down
+    // anyway and hence the timer will not fire.
+    auto * sessionManager = mInitParams.exchangeMgr->GetSessionManager();
+    if (!sessionManager)
+    {
+        return;
+    }
```

Comment:
nit: This is pure code style preference, but I do prefer the `VerifyOrReturn(sessionManager != nullptr);` style.

Reply:
Changed.